 c6450c974f4ff8520113d727d58fa74f474cd913 mount: strip multiple trailing slashes, not just one
 945fd173e9888185d68fecfb060fdfff98900ac4 mount: fix wrong long option names
 301fe50ca2576291a427de46ca6ed64d4949713e mount: ignore unknown mount options when remounting NFS mounts

---
 util-linux/mount.c |   22 +++++++++++++---------
 1 file changed, 13 insertions(+), 9 deletions(-)

diff --git a/util-linux/mount.c b/util-linux/mount.c
--- a/util-linux/mount.c
+++ b/util-linux/mount.c
@@ -241,14 +241,14 @@ static const char mount_option_str[] =
 		"union\0"
 		"bind\0"
 		"move\0"
-		"shared\0"
-		"slave\0"
-		"private\0"
-		"unbindable\0"
-		"rshared\0"
-		"rslave\0"
-		"rprivate\0"
-		"runbindable\0"
+		"make-shared\0"
+		"make-slave\0"
+		"make-private\0"
+		"make-unbindable\0"
+		"make-rshared\0"
+		"make-rslave\0"
+		"make-rprivate\0"
+		"make-runbindable\0"
 	)
 
 	// Always understood.
@@ -511,7 +511,8 @@ static int mount_it_now(struct mntent *m
 		// Remove trailing / (if any) from directory we mounted on
 
 		i = strlen(mp->mnt_dir) - 1;
-		if (i > 0 && mp->mnt_dir[i] == '/') mp->mnt_dir[i] = '\0';
+		while (i > 0 && mp->mnt_dir[i] == '/')
+			mp->mnt_dir[i--] = '\0';
 
 		// Convert to canonical pathnames as needed
 
@@ -1129,6 +1130,9 @@ static NOINLINE int nfsmount(struct mnte
 				continue;
 			case 20: // "addr" - ignore
 				continue;
+			case -1: // unknown
+				if (vfsflags & MS_REMOUNT)
+					continue;
 			}
 
 			val = xatoi_positive(opteq);
--
cgit v0.8.2.1
